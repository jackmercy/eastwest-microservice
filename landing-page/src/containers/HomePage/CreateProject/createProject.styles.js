import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import COLORS from '../../../constants/COLORS';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: { paddingTop: rem(80), paddingBottom: rem(120) },
  title: {
    fontWeight: 900,
    fontSize: rem(56),
    lineHeight: rem(72),
    margin: 0,
    transition: 'font-size .5s ease',
    textAlign: 'center',
  },
  description: {
    fontWeight: 600,
    fontSize: rem(24),
    lineHeight: rem(40),
    textAlign: 'center',
    marginTop: rem(16),
    color: COLORS.primary,
    marginBottom: 0,
    transition: 'font-size .5s ease',
  },
  content: {
    display: 'flex',
    justifyContent: 'space-between',
    marginTop: rem(80),
    boxSizing: 'border-box',
    transition: 'margin .5s ease',
  },
  form: { width: '100%', maxWidth: rem(420), minWidth: rem(320) },
  formInput: { width: '100%', '& + $formInput': { marginTop: rem(16) } },
  formButton: { width: '100%', marginTop: rem(32) },
  coverImage: {
    minWidth: 0,
    alignSelf: 'flex-start',
    display: 'block',
    marginLeft: rem(16),
    width: '100%',
    maxWidth: rem(668),
    marginTop: rem(56),
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxMd}px)`]: {
    content: { display: 'block' },
    form: { marginLeft: 'auto', marginRight: 'auto' },
    coverImage: { marginLeft: 'auto', marginRight: 'auto' },
  },
  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    root: { paddingTop: rem(48), paddingBottom: rem(80) },
    title: { fontSize: rem(32), lineHeight: rem(40) },
    description: { fontSize: rem(16), lineHeight: rem(24) },
    content: { marginTop: rem(16) },
    form: { maxWidth: '100%', minWidth: rem(200) },
    formInput: { padding: 0, '& + $formInput': { marginTop: rem(16) } },
    formButton: { maxWidth: '100%' },
    coverImage: { maxWidth: '100%', marginLeft: 0, marginTop: rem(60) },
  },
});
