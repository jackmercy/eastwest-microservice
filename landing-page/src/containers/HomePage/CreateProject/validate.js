import { formUtil } from '../../../utils';

const validations = {
  firstName: [
    { fn: formUtil.validation.require, error: 'FORM_VALIDATION.REQUIRED' },
  ],
  lastName: [
    { fn: formUtil.validation.require, error: 'FORM_VALIDATION.REQUIRED' },
  ],
  emailAddress: [
    { fn: formUtil.validation.require, error: 'FORM_VALIDATION.REQUIRED' },
    {
      fn: formUtil.validation.email,
      error: 'FORM_VALIDATION.EMAIL_INVALID',
    },
  ],
  phoneNumber: [
    { fn: formUtil.validation.require, error: 'FORM_VALIDATION.REQUIRED' },
  ],
  companyName: [
    { fn: formUtil.validation.require, error: 'FORM_VALIDATION.REQUIRED' },
  ],
  title: [
    { fn: formUtil.validation.require, error: 'FORM_VALIDATION.REQUIRED' },
  ],
  country: [
    { fn: formUtil.validation.require, error: 'FORM_VALIDATION.REQUIRED' },
  ],
};
const validate = (formValues) => formUtil.validate(formValues, validations);

export default validate;
