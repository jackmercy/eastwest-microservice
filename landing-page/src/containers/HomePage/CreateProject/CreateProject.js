import { memo, useMemo } from 'react';
import { useIntl } from 'react-intl';
import { useDispatch, useSelector } from 'react-redux';

import { Content } from '../../components';
import { Input, Button, Textarea, InputPhone } from '../../../components';
import { useFormValidation, useModalState } from '../../../hooks';
import { getLoading } from '../../../redux/ducks/loading.duck';
import { callSendNewProspectRequest } from '../../../redux/ducks/form.duck';
import { gaService } from '../../../services';

import validate from './validate';
import ModalSuccess from './ModalSuccess';
import useStyles from './createProject.styles';

const CreateProject = () => {
  const intl = useIntl();
  const classes = useStyles();
  const dispatch = useDispatch();
  const isLoading = useSelector(getLoading);
  const [openModal, toggleModal] = useModalState(false);
  const formConfigs = useMemo(
    () => [
      {
        component: Input,
        type: 'text',
        name: 'firstName',
        label: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.FIRST_NAME.LABEL',
        }),
        placeholder: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.FIRST_NAME.PLACEHOLDER',
        }),
        dataTest: 'create-project-form-input-first-name',
        value: '',
        required: true,
      },
      {
        component: Input,
        type: 'text',
        name: 'lastName',
        label: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.LAST_NAME.LABEL',
        }),
        placeholder: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.LAST_NAME.PLACEHOLDER',
        }),
        dataTest: 'create-project-form-input-last-name',
        value: '',
        required: true,
      },
      {
        component: Input,
        type: 'email',
        name: 'emailAddress',
        label: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.EMAIL_ADDRESS.LABEL',
        }),
        placeholder: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.EMAIL_ADDRESS.PLACEHOLDER',
        }),
        dataTest: 'create-project-form-input-email-address',
        value: '',
        required: true,
      },
      {
        component: InputPhone,
        name: 'phoneNumber',
        label: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.PHONE_NUMBER.LABEL',
        }),
        placeholder: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.PHONE_NUMBER.PLACEHOLDER',
        }),
        dataTest: 'create-project-form-input-phone-number',
        value: '',
        required: true,
      },
      {
        component: Input,
        type: 'text',
        name: 'companyName',
        label: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.COMPANY_NAME.LABEL',
        }),
        placeholder: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.COMPANY_NAME.PLACEHOLDER',
        }),
        dataTest: 'create-project-form-input-company-name',
        value: '',
        required: true,
      },
      {
        component: Input,
        type: 'text',
        name: 'title',
        label: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.TITLE.LABEL',
        }),
        placeholder: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.TITLE.PLACEHOLDER',
        }),
        dataTest: 'create-project-form-input-title',
        value: '',
        required: true,
      },
      {
        component: Input,
        type: 'text',
        name: 'country',
        label: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.COUNTRY.LABEL',
        }),
        placeholder: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.COUNTRY.PLACEHOLDER',
        }),
        dataTest: 'create-project-form-input-country',
        value: '',
        required: true,
      },
      {
        component: Textarea,
        name: 'description',
        label: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.COMMENT.LABEL',
        }),
        placeholder: intl.formatMessage({
          id: 'HOMEPAGE.CREATE_PROJECT.FORM.COMMENT.PLACEHOLDER',
        }),
        dataTest: 'create-project-form-textarea-comment',
        value: '',
      },
    ],
    [intl],
  );
  const {
    errors,
    formValues,
    handleResetFormValues,
    handleResetFieldError,
    handleSetFieldValue,
    translateError,
    handleSubmit,
  } = useFormValidation({}, validate);

  const handleSendEmail = () => {
    handleSubmit((formValues) =>
      dispatch(
        callSendNewProspectRequest({
          formValues,
          callback: (success) => {
            if (success) {
              toggleModal();
              gaService.event({
                action: 'submit',
                params: { form: 'prospect', formValues },
              });
            }
          },
        }),
      ));
  };

  const handleFieldChange = ({ target, name, value }) => {
    if (target) {
      handleSetFieldValue(target.name, target.value);
    } else {
      handleSetFieldValue(name, value);
    }
  };

  const handleCloseModal = () => {
    handleResetFormValues();
    toggleModal();
  };

  return (
    <>
      <Content id="create-project-form" className={classes.root}>
        <h2 className={classes.title} data-test="create-project-title">
          {intl.formatMessage({ id: 'HOMEPAGE.CREATE_PROJECT.TITLE' })}
        </h2>
        <p
          className={classes.description}
          data-test="create-project-description"
        >
          {intl.formatMessage({ id: 'HOMEPAGE.CREATE_PROJECT.DESCRIPTION' })}
        </p>
        <div className={classes.content}>
          <div className={classes.form}>
            {formConfigs.map(
              ({ component: Component, className, value, ...rest }) => (
                <Component
                  size="large"
                  key={rest.name}
                  onChange={handleFieldChange}
                  onFocus={() => handleResetFieldError(rest.name)}
                  error={translateError(errors[rest.name])}
                  disabled={isLoading}
                  className={classes.formInput}
                  value={formValues[rest.name] || value}
                  {...rest}
                />
              ),
            )}
            <Button
              size="large"
              color="primary"
              disabled={isLoading}
              className={classes.formButton}
              onClick={handleSendEmail}
              dataTest="create-project-form-button-submit"
            >
              {intl.formatMessage({
                id: 'HOMEPAGE.CREATE_PROJECT.FORM.BUTTON.SUBMIT',
              })}
            </Button>
          </div>
          <img
            alt="cover"
            loading="lazy"
            src="/static/images/homepage/cover.png"
            srcSet={[
              '/static/images/homepage/cover.png 500w',
              '/static/images/homepage/cover@2x.png 1500w',
              '/static/images/homepage/cover@3x.png 2000w',
            ].join(', ')}
            className={classes.coverImage}
            data-test="create-project-image"
          />
        </div>
      </Content>
      <ModalSuccess
        open={openModal}
        toggle={handleCloseModal}
        name={`${formValues.firstName} ${formValues.lastName}`}
      />
    </>
  );
};

export default memo(CreateProject);
