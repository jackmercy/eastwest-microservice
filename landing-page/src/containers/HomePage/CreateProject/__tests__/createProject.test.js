import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CreateProject from '../CreateProject';

describe('Testing HomePage > CreateProject', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(<CreateProject />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<CreateProject />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a title', () => {
    expect(wrapper.find('[data-test="create-project-title"]')).toHaveLength(1);
  });

  it('should have a description', () => {
    expect(
      wrapper.find('[data-test="create-project-description"]'),
    ).toHaveLength(1);
  });

  it('should have a cover image', () => {
    expect(wrapper.find('[data-test="create-project-image"]')).toHaveLength(1);
  });

  describe('Testing Contact Form', () => {
    it('should have an input first name', () => {
      expect(
        wrapper.find('[data-test="create-project-form-input-first-name"]'),
      ).toHaveLength(1);
    });

    it('should have an input last name', () => {
      expect(
        wrapper.find('[data-test="create-project-form-input-last-name"]'),
      ).toHaveLength(1);
    });

    it('should have an input email address', () => {
      expect(
        wrapper.find('[data-test="create-project-form-input-email-address"]'),
      ).toHaveLength(1);
    });

    it('should have an input phone number', () => {
      expect(
        wrapper.find('[data-test="create-project-form-input-phone-number"]'),
      ).toHaveLength(1);
    });

    it('should have an input company name', () => {
      expect(
        wrapper.find('[data-test="create-project-form-input-company-name"]'),
      ).toHaveLength(1);
    });

    it('should have an input title', () => {
      expect(
        wrapper.find('[data-test="create-project-form-input-title"]'),
      ).toHaveLength(1);
    });

    it('should have a select country', () => {
      expect(
        wrapper.find('[data-test="create-project-form-input-country"]'),
      ).toHaveLength(1);
    });

    it('should have a textarea comment', () => {
      expect(
        wrapper.find('[data-test="create-project-form-textarea-comment"]'),
      ).toHaveLength(1);
    });

    it('should have a button submit', () => {
      expect(
        wrapper.find('[data-test="create-project-form-button-submit"]'),
      ).toHaveLength(1);
    });
  });
});
