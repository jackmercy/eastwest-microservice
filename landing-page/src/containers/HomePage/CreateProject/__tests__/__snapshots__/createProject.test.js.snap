// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomePage > CreateProject should render match snapshot 1`] = `
<Fragment>
  <Memo(Content)
    className="root-0-2-1"
    id="create-project-form"
  >
    <h2
      className="title-0-2-2"
      data-test="create-project-title"
    >
      Lets start with one project.
    </h2>
    <p
      className="description-0-2-3"
      data-test="create-project-description"
    >
      You name it, We got it!
    </p>
    <div
      className="content-0-2-4"
    >
      <div
        className="form-0-2-5"
      >
        <Memo(Input)
          className="formInput-0-2-6"
          dataTest="create-project-form-input-first-name"
          error=""
          key="firstName"
          label="First name"
          name="firstName"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="First name"
          required={true}
          size="large"
          type="text"
          value=""
        />
        <Memo(Input)
          className="formInput-0-2-6"
          dataTest="create-project-form-input-last-name"
          error=""
          key="lastName"
          label="Last name"
          name="lastName"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Last name"
          required={true}
          size="large"
          type="text"
          value=""
        />
        <Memo(Input)
          className="formInput-0-2-6"
          dataTest="create-project-form-input-email-address"
          error=""
          key="emailAddress"
          label="Email address"
          name="emailAddress"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Email address"
          required={true}
          size="large"
          type="email"
          value=""
        />
        <Memo(InputPhone)
          className="formInput-0-2-6"
          dataTest="create-project-form-input-phone-number"
          error=""
          key="phoneNumber"
          label="Phone number"
          name="phoneNumber"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Phone number"
          required={true}
          size="large"
          value=""
        />
        <Memo(Input)
          className="formInput-0-2-6"
          dataTest="create-project-form-input-company-name"
          error=""
          key="companyName"
          label="Company name"
          name="companyName"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Company name"
          required={true}
          size="large"
          type="text"
          value=""
        />
        <Memo(Input)
          className="formInput-0-2-6"
          dataTest="create-project-form-input-title"
          error=""
          key="title"
          label="Title"
          name="title"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Title"
          required={true}
          size="large"
          type="text"
          value=""
        />
        <Memo(Input)
          className="formInput-0-2-6"
          dataTest="create-project-form-input-country"
          error=""
          key="country"
          label="Country"
          name="country"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Country"
          required={true}
          size="large"
          type="text"
          value=""
        />
        <Memo(Textarea)
          className="formInput-0-2-6"
          dataTest="create-project-form-textarea-comment"
          error=""
          key="description"
          label="Descibe your project"
          name="description"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Please describe your project"
          size="large"
          value=""
        />
        <Memo([object Object])
          className="formButton-0-2-7"
          color="primary"
          dataTest="create-project-form-button-submit"
          onClick={[Function]}
          size="large"
        >
          Submit
        </Memo([object Object])>
      </div>
      <img
        alt="cover"
        className="coverImage-0-2-8"
        data-test="create-project-image"
        loading="lazy"
        src="/static/images/homepage/cover.png"
        srcSet="/static/images/homepage/cover.png 500w, /static/images/homepage/cover@2x.png 1500w, /static/images/homepage/cover@3x.png 2000w"
      />
    </div>
  </Memo(Content)>
  <Memo(ModalSuccess)
    name="undefined undefined"
    open={false}
    toggle={[Function]}
  />
</Fragment>
`;
